core/templates/dev/head/pages/exploration_editor/editor_tab/StateInteraction.js

```diff
@@ -20,42 +20,23 @@
 // corresponding to an interaction id so that they can be restored if the
 // interaction is changed back while the user is still in this state. This
 // cache should be reset each time the state editor is initialized.
-oppia.factory('interactionDetailsCache', [function() {
-  var _cache = {};
-  return {
-    reset: function() {
-      _cache = {};
-    },
-    contains: function(interactionId) {
-      return _cache.hasOwnProperty(interactionId);
-    },
-    set: function(interactionId, interactionCustomizationArgs) {
-      _cache[interactionId] = {
-        customization: angular.copy(interactionCustomizationArgs)
-      };
-    },
-    get: function(interactionId) {
-      if (!_cache.hasOwnProperty(interactionId)) {
-        return null;
-      }
-      return angular.copy(_cache[interactionId]);
-    }
-  };
-}]);
+
 
 oppia.controller('StateInteraction', [
   '$scope', '$http', '$rootScope', '$modal', '$injector', '$filter',
   'alertsService', 'editorContextService', 'oppiaHtmlEscaper',
   'INTERACTION_SPECS', 'stateInteractionIdService',
   'stateCustomizationArgsService', 'editabilityService',
-  'explorationStatesService', 'graphDataService', 'interactionDetailsCache',
+  'explorationStatesService', 'graphDataService', 
+  'InteractionDetailsCacheService',
```

Comment:
You can keep this in single line only.

Reply:
There are linting error if i write it in one line 
Maximum line length exceeded